src/dotty/tools/dotc/transform/Space.scala

```diff
@@ -0,0 +1,316 @@
+package dotty.tools.dotc
+package transform
+
+import core.Types._
+import core.Contexts._
+import core.Flags
+import ast.Trees._
+import ast.tpd
+import core.Decorators._
+
+
+/** space logic for checking exhaustivity and unreachability of
+  * pattern matching.
+  */
+
+
+// space definition
+sealed trait Space
+object Empty extends Space
```

Comment:
Not much, they implement `Product0`, they have a readable `toString`, not sure what else.